File: clients/rospy/test_nodes/srv_provider.py
import rospy
from std_srvs.srv import *

def handle_request(req):
    print "Setting bool to %s"%str(req.data)
    res = SetBoolResponse()
    res.success = True
    res.message = 'Done.'
    return res

if __name__ == "__main__":
    rospy.init_node('setbool_service', log_level=rospy.DEBUG)
    s = rospy.Service('setbool', SetBool, handle_request)
    print "Ready to set bool."
    rospy.spin()

File: clients/roscpp/test_nodes/srv_provider.cpp
//
// Created by sebastian on 06.02.17.
//

#include "ros/ros.h"
#include "std_srvs/SetBool.h"

bool setbool(std_srvs::SetBool::Request  &req,
         std_srvs::SetBool::Response &res)
{
    ROS_INFO("Setting bool to request: %d", req.data);
    res.message = "Done.";
    res.success = true;
    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "setboolserver");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("setbool", setbool);
    ROS_INFO("Ready to set bool.");
    ros::spin();

    return 0;
}